### PR TITLE
Allow simulation of more htan two GPS instances

### PR DESCRIPTION
**Pull request by peterbarker**
Test with:

```
./Tools/autotest/sim_vehicle.py -v ArduPlane --gdb --debug --console  -A --serial5=sim:gps:3
```

No parameter conversion is done.

Parameters are moved to a subclass

Tested on branch for this PR: https://github.com/ArduPilot/ardupilot/pull/28581

New requirements for testing RTK moving baseline - you must specify which is your "other" GPS as well as what sort of moving baseline thing you are.

### CODE REVIEW

File: libraries/SITL/SIM_GPS.cpp [hunk 1]
```diff
@@ -30,6 +30,105 @@
 
 #include <GCS_MAVLink/GCS.h>
 
+namespace SITL {
+// user settable parameters for airspeed sensors
```
Comment (tridge) [inline libraries/SITL/SIM_GPS.cpp:34]: cut&paste?

File: libraries/SITL/SIM_GPS.cpp [hunk 2]
```diff
@@ -30,6 +30,105 @@
...
+const AP_Param::GroupInfo SIM::GPSParms::var_info[] = {
+    // @Param: ENABLE
+    // @DisplayName: GPS enable
+    // @Description: Enabled GPS
+    // @Values: 0:Disable, 1:Enable
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("ENABLE",    1, GPSParms, enabled, 0, AP_PARAM_FLAG_ENABLE),
```
Comment (tridge) [inline libraries/SITL/SIM_GPS.cpp:41]: blank line in tables

File: libraries/SITL/SIM_GPS.cpp [hunk 3]
```diff
@@ -30,6 +30,105 @@
...
+    AP_GROUPINFO("LCKTIME",  11, GPSParms, lock_time, 0),
+    // @Param: ALT_OFS
+    // @DisplayName: GPS Altitude Offset
+    // @Description: GPS Altitude Error
+    // @Units: m
+    AP_GROUPINFO("ALT_OFS",   12, GPSParms, alt_offset, 0),
+    // @Param: HDG
+    // @DisplayName: GPS Heading
+    // @Description: Enable GPS1 output of NMEA heading HDT sentence or UBLOX_RELPOSNED
+    // @Values: 0:Disabled, 1:Enabled
```
Comment (tridge) [inline libraries/SITL/SIM_GPS.cpp:107]: not boolean?

File: libraries/SITL/SITL.h [hunk 4]
```diff
@@ -155,6 +155,7 @@ class SIM {
         GPS_HEADING_HDT  = 1,
         GPS_HEADING_THS  = 2,
         GPS_HEADING_KSXT = 3,
+        GPS_HEADING_BASE = 4,  // nb.  I'm guessing, this could be ROVER!
```
Comment (tridge) [inline libraries/SITL/SITL.h:158]: needs documenting

**Review by tridge — CHANGES_REQUESTED** [review]
close!

File: libraries/SITL/SIM_GPS.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/SITL/SIM_GPS.cpp]: Fixed

File: libraries/SITL/SIM_GPS.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/SITL/SIM_GPS.cpp:42]: Fixed

File: libraries/SITL/SIM_GPS.cpp (hunk 3 above)
Comment (peterbarker) [reply libraries/SITL/SIM_GPS.cpp]: Documented

File: libraries/SITL/SITL.h (hunk 4 above)
Comment (peterbarker) [reply libraries/SITL/SITL.h]: Fixed this up - the enumeration name was correct.

**Pull request merged**